{"metadata":{"kernelspec":{"language":"python","display_name":"Python 3","name":"python3"},"language_info":{"pygments_lexer":"ipython3","nbconvert_exporter":"python","version":"3.6.4","file_extension":".py","codemirror_mode":{"name":"ipython","version":3},"name":"python","mimetype":"text/x-python"}},"nbformat_minor":4,"nbformat":4,"cells":[{"source":"<a href=\"https://www.kaggle.com/code/joshuaokolo/covid-19-dna-eda?scriptVersionId=105424522\" target=\"_blank\"><img align=\"left\" alt=\"Kaggle\" title=\"Open in Kaggle\" src=\"https://kaggle.com/static/images/open-in-kaggle.svg\"></a>","metadata":{},"cell_type":"markdown","outputs":[],"execution_count":0},{"cell_type":"markdown","source":"EDA Sample","metadata":{}},{"cell_type":"code","source":"# This Python 3 environment comes with many helpful analytics libraries installed\n# It is defined by the kaggle/python Docker image: https://github.com/kaggle/docker-python\n# For example, here's several helpful packages to load\n\nimport numpy as np # linear algebra\nimport pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n\n# Input data files are available in the read-only \"../input/\" directory\n# For example, running this (by clicking run or pressing Shift+Enter) will list all files under the input directory\n\nimport os\nfor dirname, _, filenames in os.walk('/kaggle/input'):\n    for filename in filenames:\n        print(os.path.join(dirname, filename))\n\n# You can write up to 20GB to the current directory (/kaggle/working/) that gets preserved as output when you create a version using \"Save & Run All\" \n# You can also write temporary files to /kaggle/temp/, but they won't be saved outside of the current session","metadata":{"_uuid":"8f2839f25d086af736a60e9eeb907d3b93b6e0e5","_cell_guid":"b1076dfc-b9ad-4769-8c92-a6c4dae69d19","trusted":true},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"import Bio\nfrom Bio.Seq import Seq\nfrom Bio import SeqIO\nfrom Bio.SeqUtils.ProtParam import ProteinAnalysis\nimport matplotlib.pyplot as plt\n%matplotlib inline\nimport seaborn as sns\nimport pandas as pd","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"ncov_record = SeqIO.read(\"/kaggle/input/covid19-genome-sequence/sequence.fasta\", \"fasta\")","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"ncov_dna = ncov_record.seq","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Transcription","metadata":{}},{"cell_type":"code","source":"#Protein Synthesis\nncov_mrna = ncov_dna.transcribe()\nncov_mrna","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Translation","metadata":{}},{"cell_type":"code","source":"ncov_protein = ncov_mrna.translate()\nncov_protein","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Number of Proteins","metadata":{}},{"cell_type":"code","source":"len(ncov_protein)","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Deriving Amino Acids from Protein Structures","metadata":{}},{"cell_type":"code","source":"ncov_aa = ncov_protein.split(\"*\")\nncov_aa[0:10]","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Clean List of Proteins","metadata":{}},{"cell_type":"code","source":"ncov_clean = [ str(i) for i in ncov_aa]","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"#Identify all the Proteins (chains of amino acids)\nProteins = ncov_protein.split('*')\nProteins[0:10]","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Remove chains shorter than 20 amino acids long","metadata":{}},{"cell_type":"code","source":"#Remove chains smaller than 20 amino acids long\nfor i in Proteins[:]:\n    if len(i) < 20:\n        Proteins.remove(i)","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"markdown","source":"## Protein Analysis","metadata":{}},{"cell_type":"code","source":"df = pd.DataFrame({\"amino_acids\":ncov_clean})","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"df['count'] = df['amino_acids'].str.len()","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"df.nlargest(10, \"count\")","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"#protein of interest\nsoi=df['amino_acids'].iloc[705]\nseq_oi=Seq(soi)\nseq_oi","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"#cleaning\nsoi1=''\nfor i in soi:\n    if(i!='X'):\n        soi1=soi1+i\nprint(len(soi1))","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"a=ProteinAnalysis(soi1)\na","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"MW=a.molecular_weight()\nMW","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"aromaticity=a.aromaticity()\naromaticity","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"IsoElectric=a.isoelectric_point()\nIsoElectric","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"def amino_acid_composition(protein_seq):\n    amino_acids_dict = {'Y': 0, 'A': 0, 'Q': 0, 'D': 0, 'C': 0, 'G': 0, 'V': 0, 'T': 0, 'E': 0, 'N': 0, \n                       'K': 0, 'R': 0, 'S': 0, 'I': 0, 'H': 0, 'M': 0, 'F': 0, 'L': 0, 'W': 0, 'P': 0}\n    for amino_acid in amino_acids_dict:\n        amino_acids_dict[amino_acid] = protein_seq.count(amino_acid)/len(protein_seq)*100\n    return amino_acids_dict","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"ac = amino_acid_composition(ncov_protein)\namc = pd.DataFrame.from_dict(ac, orient ='index')\namc = amc.reset_index()\namc = amc.rename(columns={\"index\": \"amino_acid\", 0: \"composition\"})","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"r = amc.sort_values(by='composition', ascending=False)\nr.style.bar(subset=[\"composition\"],color='#').background_gradient(cmap='Purples')","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"ax = sns.barplot(x=\"amino_acid\", y=\"composition\", data=r,\n                 palette=\"Blues_d\")","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"#ORF\norf_strands = []\n# generate ORFs\nfor seq in ncov_protein.split(\"STOP\"):\n    for nu in range(len(seq)-1, -1, -1):\n        if seq[nu] == \"M\":\n            orf_strands.append(seq[nu:]+\"STOP\")\n\norf_strands[0:10]","metadata":{},"execution_count":null,"outputs":[]},{"cell_type":"code","source":"len(orf_strands)\n#Conclusion: This is how I got Started with BioPython Packages","metadata":{},"execution_count":null,"outputs":[]}]}